database/postgres/dal/user.py

```diff
@@ -0,0 +1,41 @@
+from sqlalchemy import or_
+from database.models import User
+import bcrypt
+
+class UserDAL:
+    def __init__(self, session=None):
+        self.session = session
+
+    def get_user_by_email_or_username(self, email, username):
+        return self.session.query(User).filter(
+            or_(
+                User.email == email,
+                User.username == username
+            )
+        ).first()
+
+    def get_user_by_email(self, email: str) -> User:
```

Comment:
Do we need `get_user_by_email` if we already have `get_user_by_email_or_username`?

Also it doesn't add up to "email_or_username" I saw in a DTO model somehere on API level. Let's stick to one approach only (either separate username and email or joined email_or_username). Change the logic accordingly